examples/platform/silabs/efr32/rs911x/rsi_if.cpp

```diff
@@ -550,66 +540,64 @@ static void wfx_rsi_do_join(void)
+    {
+    case WFX_SEC_WEP:
+        connect_security_mode = RSI_WEP;
+        break;
+    case WFX_SEC_WPA:
+    case WFX_SEC_WPA2:
+        connect_security_mode = RSI_WPA_WPA2_MIXED;
+        break;
 #if WIFI_ENABLE_SECURITY_WPA3_TRANSITION
-        case WFX_SEC_WPA3:
-            connect_security_mode = RSI_WPA3_PERSONAL_TRANSITION;
-            break;
+    case WFX_SEC_WPA3:
+        connect_security_mode = RSI_WPA3_PERSONAL_TRANSITION;
+        break;
 #endif // WIFI_ENABLE_SECURITY_WPA3_TRANSITION
-        case WFX_SEC_NONE:
-            connect_security_mode = RSI_OPEN;
-            break;
-        default:
-            SILABS_LOG("wfx_rsi_do_join: error: unknown security type.");
-            return;
-        }
+    case WFX_SEC_NONE:
+        connect_security_mode = RSI_OPEN;
+        break;
+    default:
+        ChipLogError(DeviceLayer, "wfx_rsi_do_join: error: unknown security type.");
+        return;
+    }
 
-        SILABS_LOG("wfx_rsi_do_join: WLAN: connecting to %s, sec=%d", &wfx_rsi.sec.ssid[0], wfx_rsi.sec.security);
+    ChipLogProgress(DeviceLayer, "wfx_rsi_do_join: connecting to %s, sec=%d", &wfx_rsi.sec.ssid[0], wfx_rsi.sec.security);
 
-        /*
-         * Join the network
-         */
-        /* TODO - make the WFX_SECURITY_xxx - same as RSI_xxx
-         * Right now it's done by hand - we need something better
-         */
-        wfx_rsi.dev_state |= WFX_RSI_ST_STA_CONNECTING;
+    /*
+     * Join the network
+     */
+    /* TODO - make the WFX_SECURITY_xxx - same as RSI_xxx
+     * Right now it's done by hand - we need something better
+     */
+    wfx_rsi.dev_state |= WFX_RSI_ST_STA_CONNECTING;
 
-        if ((status = rsi_wlan_register_callbacks(RSI_JOIN_FAIL_CB, wfx_rsi_join_fail_cb)) != RSI_SUCCESS)
-        {
-            SILABS_LOG("wfx_rsi_do_join: RSI callback register join failed with status: %02x", status);
-        }
+    if ((status = rsi_wlan_register_callbacks(RSI_JOIN_FAIL_CB, wfx_rsi_join_fail_cb)) != RSI_SUCCESS)
+    {
+        ChipLogError(DeviceLayer, "wfx_rsi_do_join: rsi_wlan_register_callbacks failed: %ld", status);
+    }
 
-        /* Try to connect Wifi with given Credentials
-         * untill there is a success or maximum number of tries allowed
-         */
-        if ((status = rsi_wlan_connect_async((int8_t *) &wfx_rsi.sec.ssid[0], connect_security_mode, &wfx_rsi.sec.passkey[0],
-                                             wfx_rsi_join_cb)) != RSI_SUCCESS)
-        {
-            wfx_rsi.dev_state &= ~WFX_RSI_ST_STA_CONNECTING;
-            SILABS_LOG("wfx_rsi_do_join: rsi_wlan_connect_async failed with status: %02x on try %d", status, wfx_rsi.join_retries);
-            wfx_retry_connection(++wfx_rsi.join_retries);
-        }
+    /* Try to connect Wifi with given Credentials
+     * until there is a success or maximum number of tries allowed
+     */
+    if ((status = rsi_wlan_connect_async((int8_t *) &wfx_rsi.sec.ssid[0], connect_security_mode, &wfx_rsi.sec.passkey[0],
```

Comment:
This expects null-terminated inputs, since it doesn't take-in lengths. Is it certain the inputs are nul-terminated?

Reply:
yes, inputs have been sanitised.